kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/RequestFactory.java

```diff
@@ -66,6 +66,11 @@ public class RequestFactory {
             ApiKeys.ALTER_REPLICA_LOG_DIRS, ApiKeys.CREATE_PARTITIONS, ApiKeys.ALTER_CLIENT_QUOTAS, ApiKeys.DESCRIBE_USER_SCRAM_CREDENTIALS,
             ApiKeys.ALTER_USER_SCRAM_CREDENTIALS, ApiKeys.DESCRIBE_PRODUCERS, ApiKeys.DESCRIBE_TRANSACTIONS, ApiKeys.DESCRIBE_TOPIC_PARTITIONS);
 
+    /** The following API keys are no longer used by Kafka. They were removed by Kafka 4.0. */
+    @VisibleForTesting
+    protected static final EnumSet<ApiKeys> REMOVED_API_KEYS = EnumSet.of(ApiKeys.CONTROLLED_SHUTDOWN, ApiKeys.LEADER_AND_ISR, ApiKeys.STOP_REPLICA,
```

Comment:
Should we also track these by Kafka Version? I.e `API_KEYS_REMOVED_IN_4_0` ?

Reply:
I guess not really we aren't trying to offer support for multiple Kafka versions outside of what Kafka itself does.